zerver/webhooks/newrelic/view.py

```diff
@@ -1,169 +1,214 @@
+
+    priority = payload.get("priority").tame(check_string_in(PRIORITIES.keys()))
+    priority_symbol = PRIORITIES.get(priority, ":alert:")
+    conditions_list = payload.get("alertConditionNames", ["Unknown condition"]).tame(
+        check_list(check_string)
+    )
+    conditions = ", ".join([f"**`{c}`**" for c in conditions_list])
+    policy_list = payload.get("alertPolicyNames", ["Unknown policy"]).tame(check_list(check_string))
+    alert_policy = ", ".join([f"**`{p}`**" for p in policy_list])
+
+    message_context: dict[str, str] = {
+        "title": payload.get("title", "No title.").tame(check_string),
+        "conditions": conditions,
+        "alert_policy": alert_policy,
+        "time_created": (
+            TIME_ELEMENT_TEMPLATE.format(timestamp_data["created"])
+            if timestamp_data["created"]
+            else "N/A"
+        ),
+        "time_updated": (
+            TIME_ELEMENT_TEMPLATE.format(timestamp_data["updated"])
+            if timestamp_data["updated"]
+            else "N/A"
+        ),
+        "incident_url": payload.get("issueUrl", DEFAULT_NEWRELIC_URL).tame(check_string),
+        "total_incidents": str(payload["totalIncidents"].tame(check_int)),
+        "state": payload.get("state", "UPDATED").tame(check_string_in(INCIDENT_STATES)),
+        "status": timestamp_data["status"],
+        "priority": priority,
+        "priority_symbol": priority_symbol,
+    }
+
+    return message_context
+
+
+def format_zulip_custom_fields(payload: WildValue) -> str:
+    body_custom_field_detail: str = ""
+    zulip_custom_fields = payload.get("zulipCustomFields", {})
+
+    for key, value in zulip_custom_fields.items():
+        custom_field_name = key.capitalize()
+        try:
+            details = value.tame(
+                check_none_or(
+                    check_union(
+                        [
+                            check_int,
+                            check_float,
+                            check_string,
+                            check_list(
+                                check_none_or(check_union([check_int, check_float, check_string]))
+                            ),
+                        ]
+                    )
+                )
+            )
+            if isinstance(details, list):
+                custom_field_detail = ", ".join([f"**`{detail}`**" for detail in details])
+            else:
+                custom_field_detail = f"**{details}**"
+
+            custom_field_message = f"|{custom_field_name}: {custom_field_detail}|\n"
+            body_custom_field_detail += custom_field_message
+        except ValidationError:
+            invalid_field_message = (
+                f"|The value of **{custom_field_name}** is not a supported data type.|\n"
+            )
+            body_custom_field_detail += invalid_field_message
+    return body_custom_field_detail
+
+
+@webhook_view("NewRelic", all_event_types=INCIDENT_STATES)
```

Comment:
For consistency with other webhook views, maybe this should be `ALL_EVENT_TYPES` instead of `INCIDENT_STATES`, as `git grep all_event_types zerver/webhooks` shows this is the only outlier to that pattern.

Reply:
Switched to `ALL_EVENT_TYPES`